ddtrace/tracer/option_test.go

```diff
@@ -41,6 +41,22 @@ func withTickChan(ch <-chan time.Time) StartOption {
 // testStatsd asserts that the given statsd.Client can successfully send metrics
 // to a UDP listener located at addr.
 func testStatsd(t *testing.T, cfg *config, addr string) {
+	client := cfg.statsd
+	require.Equal(t, addr, cfg.dogstatsdAddr)
+	_, err := net.ResolveUDPAddr("udp", addr)
+	require.NoError(t, err)
+
+	client.Count("name", 1, []string{"tag"}, 1)
+	require.NoError(t, client.Close())
+}
+
+func TestStatsdUDPConnect(t *testing.T) {
+	defer func(old string) { os.Setenv("DD_DOGSTATSD_PORT", old) }(os.Getenv("DD_DOGSTATSD_PORT"))
+	os.Setenv("DD_DOGSTATSD_PORT", "8111")
+	testStatsd(t, newConfig(), net.JoinHostPort(defaultHostname, "8111"))
+	cfg := newConfig()
+	addr := net.JoinHostPort(defaultHostname, "8111")
+
```

Comment:
This is a general fix. It's inconvenient to actually listen on every UDP port we're testing. For the most part we just want to make sure that things are configured as we expect, and then for one specific port we make sure we can listen, connect, and send stats.

This also helps prevent tests from failing locally when there is an agent running.